src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class AckHandler : IDisposable
+    {
+        private readonly ConcurrentDictionary<string, AckInfo> _acks = new ConcurrentDictionary<string, AckInfo>();
+        private readonly Timer _timer;
+        private readonly TimeSpan _ackThreshold = TimeSpan.FromSeconds(10);
+        private readonly TimeSpan _ackInterval = TimeSpan.FromSeconds(5);
+
+        public AckHandler()
+        {
+            _timer = new Timer(_ => CheckAcks(), state: null, dueTime: _ackInterval, period: _ackInterval);
```

Comment:
Consider not capturing the ExecutionContext, see https://github.com/aspnet/AspNetCore/blob/master/src/SignalR/server/StackExchangeRedis/src/Internal/AckHandler.cs#L22